Packages/com.unity.inputsystem/Documentation~/Debugging.md

```diff
@@ -28,41 +23,45 @@ The debugger shows a tree breakdown of the state of the input system.
 
->NOTE: For an alternative way to debug/visualize controls and/or devices, see [Visualizing Controls](#visualizing-controls).
+Finally, in the bottom of the window, you can inspect the event stream coming from the device. Any [Input Event](Events.md) generated by the device will be shown here. You can double-click any event in the list to inspect the full device state at the time of the event.
 
 ### Debugging Actions
 
->NOTE: For an alternative way to debug/visualize actions, see [Visualizing Actions](#visualizing-actions).
+Any active [actions](Actions.md) are listed in the Actions list in the debugger window (only shown if any actions are active and the editor is in play mode). If an action has actively bound controls, you can click the arrow next to the action to see them in the list. This is useful to debug whether your bindings correctly map to the controls you want them to bind to. See ["Binding Resolution"](ActionBindings.md#binding-resolution) for more info on how bindings are mapped to controls.
 
 ### Debugging Users/PlayerInput
 
-When there are multiple `InputUser` instances &ndash; each `PlayerInput` will implicitly create one &ndash;, the debugger will list each user along with its paired devices and active actions.
+When there are multiple `InputUser` instances &ndash; each `PlayerInput` will implicitly create one &ndash;, the debugger will list each user along with its paired devices and active actions. The listed devices and actions work the same way as those shown in the [devices](#debugging-devices) and [actions](#debugging-actions) lists in the debugging window.
 
 IMAGE
 
-## Input Visualizers
+### Debugging Layouts
 
-To debug specific problems it is often helpful to see activity over time and presented in graphical form.
+The [`Layouts`](Layouts.md) list in the Debugger window shows a breakdown of all registered [control and device layouts](Layouts.md). This is essentially the database of supported hardware and the knowledge of how to represent a given piece of input hardware. This is mostly handy when you want to [create a new device mapping](HowDoI.md#-create-my-own-custom-devices), and want to see how it gets represented by the input system.
 
->NOTE: The input visualizer components are only available in the editor and in development players.
+### Debugging Remotely
 
-### Visualizing Controls
+You can connect input debugger to a player running on a device. This makes input activity from the player observable in the editor. The mechanism used by this is Unity's `PlayerConnection`, i.e. the same mechanism by which the Unity profiler can be connected to a player.
 
-### Visualizing Actions
+>NOTE: At the moment, debugging input in players is restricted to seeing devices and events from connected players. There is no support yet for seeing other input-related data from players.
```

Comment:
Might be worth adding something like "such as actions and input users." for extra explicitness.

Reply:
Done